src/core/Akka.Remote/Transport/Helios/HeliosTransport.cs

```diff
@@ -208,150 +217,255 @@ private TransportType InternalTransport
+                    throw new ConfigurationException($"Unknown local address type {ServerChannel.LocalAddress}");
+                LocalAddress = addr;
+                // resume accepting incoming connections
+#pragma warning disable 4014 // we WANT this task to run without waiting
+                AssociationListenerPromise.Task.ContinueWith(result => ServerChannel.Configuration.AutoRead = true,
+                    TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion);
+#pragma warning restore 4014
+
+
+                return Tuple.Create(addr, AssociationListenerPromise);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Failed to bind to {0}; shutting down Helios transport.", listenAddress);
+                try
+                {
+                    await Shutdown();
+                }
+                catch
+                {
+                    // ignore errors occurring during shutdown
+                }
+                throw;
+            }
         }
 
         public override async Task<AssociationHandle> Associate(Address remoteAddress)
         {
-            if (!ServerChannel.IsOpen())
-               throw new HeliosConnectionException(ExceptionType.NotOpen, "Transport is not open");
+            if (!ServerChannel.IsOpen)
+                throw new HeliosConnectionException(ExceptionType.NotOpen, "Transport is not open");
 
             return await AssociateInternal(remoteAddress);
         }
 
-        public override Task<bool> Shutdown()
+        public override async Task<bool> Shutdown()
         {
-            return Task.Run(() =>
+            try
             {
-                try
-                {
-                    foreach (var channel in ConnectionGroup)
-                    {
-                        channel.StopReceive();
-                        channel.Dispose();
-                    }
-                }
-                finally
+                var tasks = new List<Task>();
+                foreach (var channel in ConnectionGroup)
                 {
-                    // free all of the connection objects we were holding onto
-                    ConnectionGroup.Clear();
+                    tasks.Add(channel.CloseAsync());
                 }
-                
-                return true;
-            });
+                var all = Task.WhenAll(tasks);
+                await all;
+
+                var server = ServerChannel?.CloseAsync() ?? TaskEx.Completed;
+                await server;
+
+                return all.IsCompleted && server.IsCompleted;
+            }
+            finally
+            {
+                // free all of the connection objects we were holding onto
+                ConnectionGroup.Clear();
+#pragma warning disable 4014 // shutting down the worker groups can take up to 10 seconds each. Let that happen asnychronously.
+                _clientEventLoopGroup.ShutdownGracefullyAsync();
```

Comment:
We shut down the event loops separately and only after the `IChannel`s themselves are shutdown - that way we can avoid a race condition where the event loops abort the threads the channels are using to perform cleanup.